testing/integration/beat_receivers_test.go

```diff
@@ -79,73 +75,77 @@ func TestAgentMonitoring(t *testing.T) {
 		Force:          true,
 	}
 
+	// prepare the policy and marshalled configuration
+	policyCtx, policyCancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(5*time.Minute))
+	t.Cleanup(policyCancel)
+
+	// 1. Create and install policy with just monitoring
+	createPolicyReq := kibana.AgentPolicy{
+		Name:        fmt.Sprintf("%s-%s", t.Name(), uuid.Must(uuid.NewV4()).String()),
+		Namespace:   info.Namespace,
+		Description: fmt.Sprintf("%s policy", t.Name()),
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policyResponse, err := info.KibanaClient.CreatePolicy(policyCtx, createPolicyReq)
+	require.NoError(t, err, "error creating policy")
+
+	// 2. Download the policy, add the API key
+	downloadURL := fmt.Sprintf("/api/fleet/agent_policies/%s/download", policyResponse.ID)
+	resp, err := info.KibanaClient.Connection.SendWithContext(policyCtx, http.MethodGet, downloadURL, nil, nil, nil)
+	require.NoError(t, err, "error downloading policy")
+	policyBytes, err := io.ReadAll(resp.Body)
+	require.NoError(t, err, "error reading policy response")
+	defer resp.Body.Close()
```

Comment:
Could we extract this + the `PolicyStruct` type + the parsing of `policyBytes` into a `PolicyStruct` below into https://github.com/elastic/elastic-agent-libs/tree/main/kibana so the code here is simply making a call to something like `info.KibanaClient.GetPolicyDownload(policyResponse.ID)`?

Reply:
Never mind, while the above would be desirable, I see now that you are just moving this code around from what it was already.

Reply:
Honestly, I'm not convinced any of the Kibana policy stuff in this test is even necessary vs just explicitly writing the standalone agent config we want. But like you said, I'm just moving things around here.